plugins/kvscheduler/api/value_status.proto

```diff
@@ -0,0 +1,63 @@
+syntax = "proto3";
+
+package api;
+
+enum ValueState {
+    // ValueState_NONEXISTENT is assigned to value that was deleted or has never
+    // existed.
+    NONEXISTENT = 0;
+
+    // ValueState_CONFIGURED marks value defined by NB and successfully configured.
+    CONFIGURED = 1;
+
+    // ValueState_RETRIEVED marks value not managed by NB, instead created
+    // automatically or externally in SB.
+    RETRIEVED = 2;
+
+    // ValueState_PENDING represents (NB) value that cannot be configured yet
+    // due to missing dependencies.
+    PENDING = 3;
+
+    // ValueState_INVALID represents (NB) value that will not be configured
+    // because it either cannot be unmarshalled or it has an invalid content
+    // as declared by the Validate method of the associated descriptor.
+    // The corresponding error is stored in the <error> field of the value status.
+    INVALID = 4;
+
+    // ValueState_UNIMPLEMENTED marks value received from NB that cannot
+    // be configured because there is no registered descriptor associated with it.
+    UNIMPLEMENTED = 5;
+
+    // ValueState_FAILED marks (NB) value for which the last executed operation
+    // returned an error. The error is stored in the <error> field of the value
+    // status.
+    FAILED = 6;
+}
+
+message ValueError {
```

Comment:
This should rather be named `ValueDetail`, because the value state might be in non-failed state like `PENDING` and `ValueDetail` could describe the missing dependency, or the state could be in `INVALID` state and this could describe what part of it is not valid.